src/pages/SwapNew/components/CurrencyItem/CurrencyAmount.tsx

```diff
@@ -0,0 +1,104 @@
+import { CurrencyAmount as CurrencyAmountType } from '@swapr/sdk'
+
+import { ChangeEvent, useState, useMemo, useCallback, useEffect } from 'react'
+import styled from 'styled-components'
+
+import { escapeRegExp, normalizeInputValue } from '../../../../utils'
+import { debounce } from '../../../../utils/debounce'
+import { TEXT_COLOR_PRIMARY } from '../../constants'
+import { FontFamily } from '../../constants'
+
+const inputRegex = RegExp(`^\\d*(?:\\\\[.])?\\d*$`)
+
+type CurrencyAmountProps = {
+  value: string | number
+  onUserInput: (value: string) => void
+  setIsMaxAmount: (value: boolean) => void
+  maxAmount?: CurrencyAmountType
+  isInputPanelDisabled?: boolean
+}
+
+export function CurrencyAmount({
+  value,
+  onUserInput,
+  setIsMaxAmount,
+  maxAmount,
+  isInputPanelDisabled,
+}: CurrencyAmountProps) {
+  const [inputValue, setInputValue] = useState(value)
+
+  const handleOnUserInput = useCallback(
+    (value: string) => {
+      if (maxAmount?.toExact() === value) setIsMaxAmount(true)
+      else setIsMaxAmount(false)
+
+      onUserInput(value)
+    },
+    [maxAmount, onUserInput, setIsMaxAmount]
+  )
+
+  const debouncedUserInput = useMemo(() => {
+    return debounce(handleOnUserInput, 250)
+  }, [handleOnUserInput])
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    event.stopPropagation()
+
+    const parsedValue = event.target.value.replace(/,/g, '.')
+
+    if (parsedValue === '' || inputRegex.test(escapeRegExp(parsedValue))) {
+      setInputValue(normalizeInputValue(parsedValue))
+      debouncedUserInput(normalizeInputValue(parsedValue))
+    }
+  }
+
+  useEffect(() => {
+    if (inputValue !== value) {
+      setInputValue(value)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value])
+
+  return (
+    <StyledInput
+      value={inputValue}
+      onChange={handleInputChange}
+      type="number"
+      inputMode="decimal"
+      autoComplete="off"
+      autoCorrect="off"
+      pattern="^[0-9]*[.,]?[0-9]*$"
+      placeholder="0.0"
+      minLength={1}
+      maxLength={79}
+      spellCheck="false"
+      disabled={isInputPanelDisabled}
+    />
+  )
+}
+
+const StyledInput = styled.input`
+  width: 100%;
+  height: 34px;
+  line-height: 34px;
+  font-size: 28px;
+  ${FontFamily}
+  font-weight: 600;
+  letter-spacing: 0.02em;
+  /* text-shadow: 0px 0px 12px rgba(255, 255, 255, 0.14); */
+  color: ${TEXT_COLOR_PRIMARY};
```

Comment:
I think its best to extend the existing theme for all the colors and styling instead of creating a new styled file



and create a swapBox in the theme for specific separation in `src/theme/styled.d.ts`
```

export interface SwapBoxIndicatorColors {
  positive: string
  warning: string
  negative: string
  undefined: string
}


export interface SwapBox {
  indicatorColors: SwapBoxIndicatorColors
  indicatorBgColors: SwapBoxIndicatorColors
  spacing: string
  textPrimary: string
  textSecondary: string
}
```
and extend `DefaultTheme`

```
swapBox: {
      indicatorColors: {
        positive: '#0E9F6E',
        warning: '#F2994A',
        negative: '#F02E51',
        undefined: '#8780BF',
      },
      indicatorBgColors: {
        positive: '#0e9f6e14',
        warning: '#f2994a14',
        negative: '#f02e5114',
        undefined: '#8780bf14',
      },
      spacing: '6px',
      textPrimary: '#fff',
      textSecondary: '#8780bf',
    },

```
so can be used like this 

```
  color: ${({ theme, color }) => theme.swapBox.indicatorColors[color]};
  background: ${({ theme, color }) => theme.swapBox.indicatorBgColors[color]};
```

Reply:
I am not yet sure about the place where this'll go but I'll definitely explore the option and consider updating the pull request. Thanks @wixzi. 🙏